### PR TITLE
Danderson/v6 lazy cfg

### PR DESCRIPTION
**Pull request by danderson**
WIP. Still needs docstring fixups, and lazy cfg cleanup code hasn't been adjusted.

May also refactor a bit of the send activity stuff to be more in tstun and involve fewer anonymous functions... But still thinking through the API implications.

### CODE REVIEW

**Comment by danderson** [comment]
Thanks github for pulling all the comments out of line...

Re: AllowedIPs, my plan is to enable IPv6 with a new MapVersion, so code that can't handle v6 will continue to work as before.

Ack on the other two comments, will incorporate.

**Comment by danderson** [comment]
The BitLen func doesn't help in this particular case, because wgengine is operating on wgcfg.CIDRs at that point in the code. Possibly worth looking at doing more work before the conversion into a wgcfg instead, but that's for another change.

File: net/packet/ip6.go [hunk 1]
```diff
@@ -16,13 +16,17 @@ type IP6 struct {
 	Hi, Lo uint64
 }
 
+// IP6FromRaw16 converts a raw 16-byte IPv6 address to an IP6.
+func IP6FromRaw16(ip [16]byte) IP6 {
```
Comment (bradfitz) [inline net/packet/ip6.go:20]: Actually, @josharian, I was wrong. Not three, but 4 IP types 😂 

@josharian is talking about deleting the `wgcfg` package and its type.

See also: https://github.com/inetaf/netaddr/issues/53

File: wgengine/tstun/tun.go [hunk 2]
```diff
@@ -136,8 +137,9 @@ func WrapTUN(logf logger.Logf, tdev tun.Device) *TUN {
 // destination (the map keys).
 //
 // The map ownership passes to the TUN. It must be non-nil.
-func (t *TUN) SetDestIPActivityFuncs(m map[packet.IP4]func()) {
-	t.destIPActivity.Store(m)
+func (t *TUN) SetDestIPActivityFuncs(m4 map[packet.IP4]func(), m6 map[packet.IP6]func()) {
```
Comment (bradfitz) [inline wgengine/tstun/tun.go:140]: If we did https://github.com/inetaf/netaddr/issues/53 first, could this simplify down to just one map again?

**Comment by danderson** [comment]
I'm going to merge this as-is, then do a separate series today to switch to the new and improved netaddr types. That way if our unsafe shenanigans backfire and require a rollback of netaddr, we have a rollback point in tailscale/tailscale as well.

**Pull request merged**